### PR TITLE
added missing dependencies icu and liblzma

### PR DESCRIPTION
**Pull request by vgauthier**
I encountered an issue of missing while installing python-igraph on osx. So I changed in the meta.yaml the two missing dependencies:
- backports.lzma
- icu support

### CODE REVIEW

**Comment by sodre** [comment]
I will look into it tonight.

**Comment by vgauthier** [comment]
the dependency for backports.lzma is not available for py34, so I skipped the build for py34

File: recipe/meta.yaml [hunk 1]
```diff
@@ -23,10 +23,17 @@ requirements:
     - python
     - setuptools
     - toolchain
-
+    - backports.lzma
+    - vc 9  # [win and py27]
```
Comment (sodre) [inline recipe/meta.yaml]: These are not needed since we are not building for Windows *yet*.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -23,10 +23,17 @@ requirements:
...
+    - vc 10  # [win and py34]
+    - vc 14  # [win and py>=35]
   run:
     - igraph 0.7.1
     - python
+    - backports.lzma
+    - vc 9  # [win and py27]
```
Comment (sodre) [inline recipe/meta.yaml]: ditto.

File: recipe/meta.yaml [hunk 3]
```diff
@@ -23,10 +23,17 @@ requirements:
     - python
     - setuptools
     - toolchain
-
+    - backports.lzma
```
Comment (sodre) [inline recipe/meta.yaml]: is backports needed during build or only runtime?

Can you add the unit test that shows how it failed?

File: recipe/meta.yaml [hunk 4]
```diff
@@ -12,9 +12,9 @@ source:
   sha256: {{ sha256 }}
 
 build:
-  number: 1
+  number: 2
   script: python setup.py install --single-version-externally-managed --record record.txt
-  skip : true  # [win]
+  skip : true  # [win or py34]
```
Comment (sodre) [inline recipe/meta.yaml]: Remove the py34 skip and rebase once I have re rendered the recipe.

File: recipe/meta.yaml [hunk 5]
```diff
@@ -12,9 +12,9 @@ source:
   sha256: {{ sha256 }}
 
 build:
-  number: 1
+  number: 2
```
Comment (sodre) [inline recipe/meta.yaml]: This will need to get bumped to 3 after you rebase.

**Comment by vgauthier** [comment]
It seems there are two issues with osx: 

The first is related the missing icu library when I do import graph

``python
>>> import igraph
ImportError: dlopen(/usr/local/anaconda/envs/py36/lib/python3.6/site-packages/igraph/_igraph.cpython-36m-darwin.so, 2): Library not loaded: @rpath/libicui18n.58.dylib
  Referenced from: /usr/local/anaconda/envs/py36/lib/libigraph.0.dylib
  Reason: image not found
``

The secon issue append only with py36 when I do the import of the igraph library
``
>>> import igraph
 ImportError: dlopen(/usr/local/anaconda/envs/py36/lib/python3.6/site-packages/igraph/_igraph.cpython-36m-darwin.so, 2): Symbol not found: _PySlice_AdjustIndices
    Referenced from: /usr/local/anaconda/envs/py36/lib/python3.6/site-packages/igraph/_igraph.cpython-36m-darwin.so
    Expected in: flat namespace
   in /usr/local/anaconda/envs/py36/lib/python3.6/site-packages/igraph/_igraph.cpython-36m-darwin.so
``

File: recipe/meta.yaml (hunk 4 above)
Comment (vgauthier) [inline recipe/meta.yaml]: right

File: recipe/meta.yaml (hunk 2 above)
Comment (vgauthier) [inline recipe/meta.yaml]: right

File: recipe/meta.yaml (hunk 5 above)
Comment (vgauthier) [inline recipe/meta.yaml]: Done

**Comment by sodre** [comment]
Hi @vgauthier,
Can you take a look at #4, I added your test case directly to the existing code base and everything seems to be fine even without your other patches.  I also tested on my box macOS Sierra 10.12.4 and it igraph works fine.

Can you try `conda create -n "igraph-env" -c condo-forge python-igraph` and see if that works?
If it is still failing, can you send the output of `conda list`  to me.

**Comment by vgauthier** [comment]
Hi, 

You right, It seems that some package in my conda list come from another channel than conda-forge. This caused some issue with dynamically linked libraries such as libxml. With, the correct dependency coming from conda-forge I do not have this issue anymore, my bad.

However, I saw that pycairo that is used by igraph is now available on conda-forge. I think you can include it in the dependency list.

Best

**Comment by sodre** [comment]
@vgauthier is do you have a test case for pycairo? you can just point me in the right direction on how to test and I will add it.

**Comment by vgauthier** [comment]
@sodre I can think about something like that to test the plotting backend of igraph

```Python
import igraph as ig
import matplotlib.pylab as plt

G = ig.Graph(directed=False)
G.add_vertices(n=4)
G.add_edges([(0, 1),
             (0, 2),
             (0, 3),
             (1, 2),
             (1, 3),
             (2, 3)])
G['layout'] = G.layout_fruchterman_reingold()
ig.plot(G, "test.pdf")
```

**Comment by sodre** [comment]
@vgauthier it looks like the lzma and icu issues were not caused by igraph. I am closing this PR.

I opened another PR for the pycairo issue and asked you to be a reviewer. If it looks good to you I will merge it.

**Pull request closed without merging**